unity-renderer/Assets/DCLPlugins/ECS7/Systems/EngineInfoSystem/ECSEngineInfoSystem.cs

```diff
@@ -0,0 +1,54 @@
+using DCL.ECS7;
+using DCL.ECS7.InternalComponents;
+using DCL.ECSComponents;
+using DCL.ECSRuntime;
+using DCL.Models;
+using UnityEngine;
+
+namespace ECSSystems.ECSEngineInfoSystem
+{
+    /// <summary>
+    /// This system updates every scene's root entity EngineInfo component on every frame
+    /// </summary>
+    public class ECSEngineInfoSystem
+    {
+        private readonly IECSComponentWriter componentWriter;
+        private readonly IInternalECSComponent<InternalEngineInfo> internalEngineInfo;
+
+        public ECSEngineInfoSystem(
+            IECSComponentWriter componentWriter,
+            IInternalECSComponent<InternalEngineInfo> internalEngineInfo)
+        {
+            this.componentWriter = componentWriter;
+            this.internalEngineInfo = internalEngineInfo;
+        }
+
+        public void Update()
+        {
+            int currentEngineFrameCount = Time.frameCount;
+            float currentEngineRunTime = Time.realtimeSinceStartup;
+
+            // Scenes internal EngineInfo component is initialized at SceneStateHandler.InitializeEngineInfoComponent()
+            var componentGroup = internalEngineInfo.GetForAll();
+
+            int entitiesCount = componentGroup.Count;
+            for (int i = 0; i < entitiesCount; i++)
+            {
+                var entity = componentGroup[i].value.entity;
+                var scene = componentGroup[i].value.scene;
+                var model = componentGroup[i].value.model;
+
+                componentWriter.PutComponent(
+                    entity.scene,
+                    scene.GetEntityById(SpecialEntityId.SCENE_ROOT_ENTITY),
+                    ComponentID.ENGINE_INFO,
+                    new PBEngineInfo()
+                    {
+                        TickNumber = model.SceneTick,
+                        FrameNumber = (uint)currentEngineFrameCount,
```

Comment:
shouldn't `FrameNumber` be the frames count since scene was loaded?

Reply:
in [the protocol](https://github.com/decentraland/protocol/blob/main/proto/decentraland/sdk/components/engine_info.proto) it says "frame counter of the engine" unlike the other 2 that specify that it's scene-related, what do we want to do with this? I have no problem adding a starting frame count value in the internal engine info component and using that to calculate the scene frame count

Reply:
in [the protocol](https://github.com/decentraland/protocol/blob/main/proto/decentraland/sdk/components/engine_info.proto) it says "frame counter of the engine" unlike the other 2 that specify that it's scene-related, what do we want to do with this? I have no problem adding a starting frame count value in the internal engine info component and using that to calculate the scene frame count

Reply:
It still would be "frame counter of the engine" if it's relative to the scene load.
Imagine you scene just load and you get a `FrameNumber` of 1 million because you were walking around the world, it's kind of odd

Reply:
in any case, let's ask @menduz :)

Reply:
All counters start relative to the scene, no absolute numbers around here

Reply:
Cool, done!